### PR TITLE
Updated events for erc4626 linear pool factory

### PR DESCRIPTION
**Pull request by baileyspraggins**
# Description

Adds support for ERC462Linear V3.
This is being done before factories are deployed so as to reduce lead time between deployment and pools being indexed on the subgraph. In case factory addresses end up not matching the ones provided here, a new sync will be required

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Dependency changes
- [ ] Code refactor / cleanup
- [ ] Documentation or wording changes
- [ ] Other

## How should this be tested?

N/A (factories not deployed)

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have requested at least 1 review (If the PR is significant enough, use best judgement here)
- [x] I have commented my code where relevant, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by baileyspraggins** [comment]
Events within the ERC4626 Linear Pool Factory: 
```
    // This event allows off-chain tools to differentiate between different protocols that use this factory
    // to deploy Erc4626 Linear Pools.
    event Erc4626LinearPoolCreated(address indexed pool, uint256 indexed protocolId);

    // Record protocol ID registrations.
    event Erc4626LinearPoolProtocolIdRegistered(uint256 indexed protocolId, string name);

File: src/mappings/poolFactory.ts [hunk 1]
```diff
@@ -250,10 +251,25 @@ export function handleNewAaveLinearPoolV3(event: AaveLinearPoolCreated): void {
...
+  const poolCreatedEvent = new PoolCreated(
+    event.address,
+    event.logIndex,
+    event.transactionLogIndex,
+    event.logType,
+    event.block,
+    event.transaction,
+    [event.parameters[0]]
+  );
+  handleNewLinearPool(poolCreatedEvent, PoolType.ERC4626Linear, 1, event.params.protocolId.toI32());
```
Comment (markusbkoch) [inline src/mappings/poolFactory.ts:270]: pooltype version here should be 3
```suggestion
  handleNewLinearPool(poolCreatedEvent, PoolType.ERC4626Linear, 3, event.params.protocolId.toI32());
```

File: manifest.template.yaml [hunk 2]
```diff
@@ -683,6 +683,39 @@ dataSources:
...
+          file: ./abis/Vault.json
+        - name: ERC20
+          file: ./abis/ERC20.json
+        - name: WeightedPool
+          file: ./abis/WeightedPool.json
+        - name: ERC4626LinearPoolV3Factory
+          file: ./abis/ERC4626LinearPoolV3Factory.json
+        - name: LinearPool
+          file: ./abis/LinearPool.json
+        - name: ERC4626LinearPoolV3Factory
```
Comment (markusbkoch) [inline manifest.template.yaml:713]: I think you meant `ERC4626LinearPoolV3Pool` but also I'm not sure this is needed, is it @mendesfabio ?

File: src/mappings/poolFactory.ts [hunk 3]
```diff
@@ -250,10 +251,25 @@ export function handleNewAaveLinearPoolV3(event: AaveLinearPoolCreated): void {
   handleNewLinearPool(poolCreatedEvent, PoolType.AaveLinear, 3, event.params.protocolId.toI32());
 }
 
+
 export function handleNewERC4626LinearPool(event: PoolCreated): void {
   handleNewLinearPool(event, PoolType.ERC4626Linear);
 }
 
+// TODO: Import Erc4626LinearPoolCreated once abi is addded
```
Comment (markusbkoch) [inline src/mappings/poolFactory.ts:259]: Keep in mind that for this we don't need the complete and finalized ABI, just a mock ABI file with the Erc4626LinearPoolCreated event

**Comment by baileyspraggins** [comment]
Missing startBlocks and Goerli Address

File: manifest.template.yaml [hunk 4]
```diff
@@ -683,6 +683,39 @@ dataSources:
...
+        - name: ERC20
+          file: ./abis/ERC20.json
+        - name: WeightedPool
+          file: ./abis/WeightedPool.json
+        - name: ERC4626LinearPoolV3Factory
+          file: ./abis/ERC4626LinearPoolV3Factory.json
+        - name: LinearPool
+          file: ./abis/LinearPool.json
+        - name: ERC4626LinearPoolV3Pool
+          file: ./abis/ERC4626LinearPoolV3Pool.json
```
Comment (markusbkoch) [inline manifest.template.yaml:714]:
```suggestion
          file: ./abis/ERC4626LinearPoolV3.json
```

**Review by markusbkoch — CHANGES_REQUESTED** [review]
Thank you. Another couple of change requests:
- add the `.json` extension to file `ERC4626LinearPoolV3Factory`
- set the PR target to `staging-dev` instead of `master`
- set `startBlock` to the most recent block on each network
- fix lint errors (`yarn lint --fix`)

The Goerli address is not really a blocker for this -  we can merge that in later with no impact to the sync process of the production networks

File: networks.yaml [hunk 5]
```diff
@@ -255,6 +261,9 @@ arbitrum:
   AaveLinearPoolV3Factory:
     address: "0xa2D801064652A269D92EE2A59F3261155ec66830"
     startBlock: 44470235
+  ERC4626LinearPoolV3Factory:
+    address: "0xa3B9515A9c557455BC53F7a535A85219b59e8B2E"
+    startBlock: 71912978
```
Comment (markusbkoch) [inline networks.yaml:266]:
```suggestion
    startBlock: 57735268
```

Comment (markusbkoch) [reply networks.yaml:266]: source: https://arbiscan.io/

**Review by markusbkoch — CHANGES_REQUESTED** [review]

**Pull request merged**